test/test_troubleshooting.py

```diff
@@ -0,0 +1,28 @@
+import subprocess
+import pytest
+from .test_projects import TestProject
+from . import utils
+
+so_file_project = TestProject()
+
+so_file_project.files['libnothing.so'] = ''
```

Comment:
Oh, wow. I hadn't realized it would be so easy to test, but of course! That's amazing!